src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiParameterGenerator.cs

```diff
@@ -134,7 +134,10 @@ public static IList<OpenApiParameter> CreateParameters(this ODataContext context
             if (keys.Count() == 1)
             {
                 string keyName = keys.First().Name;
-                if (context.Settings.PrefixEntityTypeNameBeforeKey)
+
+                // If dictionary parameterNameMapping is defined, there's no need of setting the
+                // keyName, we will retrieve this from the dictionary key.
+                if (context.Settings.PrefixEntityTypeNameBeforeKey && parameterNameMapping == null)
```

Comment:
question: this completely ignores the context setting so that the name of the OpenApiParameter is not prefixed with the entity type name.  Shouldn't we still add the prefix to the value from the dictionary once we retrieve it?

Reply:
The prefixed key gets set here in the `keyNamesMapping` dictionary in this method: https://github.com/microsoftgraph/OpenAPI.NET.OData/blob/97595daf2f60b983c2c8b34642a669c6a7893515/src/Microsoft.OpenApi.OData.Reader/Edm/ODataKeySegment.cs#L91-L95

Therefore, this evaluation 
https://github.com/microsoftgraph/OpenAPI.NET.OData/blob/97595daf2f60b983c2c8b34642a669c6a7893515/src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiParameterGenerator.cs#L147

will use the key that was created here, from the previous method: https://github.com/microsoftgraph/OpenAPI.NET.OData/blob/97595daf2f60b983c2c8b34642a669c6a7893515/src/Microsoft.OpenApi.OData.Reader/Edm/ODataKeySegment.cs#L89

if the `parameterNameMapping` dictionary is `!null`

Reply:
got it.